crates/cheatnet/src/state.rs

```diff
@@ -18,6 +18,37 @@ use starknet_api::{
 };
 use std::collections::HashMap;
 
+pub struct CustomStateReader(pub Box<dyn StateReader>);
```

Comment:
StateReaderProxy/Router maybe would be a better name?

Reply:
What is the performance overhead of keeping it on the heap? I assume nothing significant but I wanted to raise it so you can think about it.

Reply:
I don't think it will be much more heavier than the one without forking